File: 프로그래머스/lv2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    var answer = 0;
    let print = 0;
    while (true) {
        let task = priorities.shift();
        let most = true;
        priorities.forEach(el=>{
            if(el>task) most = false;
        });
        if(most) {
            answer++;
            if(location===0) break;
            else location--;
        } else {
            priorities.push(task);
            if (location===0) location=priorities.length-1;
            else location--;
        }
    }
    return answer;
}

File: 프로그래머스/lv2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        int qp = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>((a,b) -> b-a);
        for(int p : priorities) {
            pq.offer(p);
        }

        while(true) {
            if(pq.peek()==priorities[qp]) {
                answer++;
                if(location==qp) break;
                priorities[qp] = -1;
                pq.poll();
            }
            qp = (qp+1)%priorities.length;
        }
        return answer;
    }
}